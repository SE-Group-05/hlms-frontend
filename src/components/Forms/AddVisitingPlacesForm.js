import React, { useState } from "react";
import {
  Input, Text, Stack, FormControl, Button, Divider, Heading, Box, Alert, AlertIcon, Textarea, InputGroup, FormHelperText, RadioGroup, Radio,
  NumberInput,
  NumberInputField,
  NumberInputStepper,
  NumberIncrementStepper,
  NumberDecrementStepper,
  Checkbox,
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalFooter,
  ModalBody,
  ModalCloseButton,
  useDisclosure,
  Center,
  VStack
} from "@chakra-ui/react";

import { Link } from 'react-router-dom';
import Map from "../Admin/Map";
import { isNumeric, isRequired, isRequiredArr } from '../../utils/Validation2';
import { visitingPlaceService } from "../../services/visitingplaceService";
import { authHeader } from "../../services/_helpers/authheader";
import { CheckCircleIcon, QuestionIcon } from '@chakra-ui/icons'
const BASE_URL = 'http://localhost:5000'


function VisitingPlaceForm() {
  const [data, setData] = useState({
    placeName: '',
    description: '',
    latitude: '',
    longitude: '',
    method: '',
  });

  const { isOpen, onOpen, onClose } = useDisclosure()
  const [distance, setDistance] = useState(0)
  const [timetoreach, setTimetoreach] = useState(0)

  const [tmethods, setTmethods] = useState([]);

  const [hasError, setHasError] = useState(false);
  const [confirm, setConfirm] = useState(false);

  const [validation, setValidation] = useState({
    validation: [false, false, false, false, false, false, false],
    errorMessage: ['', '', '', '', '', '', '']
  });

  const showResult = (result) => {
    onOpen();
  }

  const validateBeforeSubmit = () => {
    console.log(tmethods);
    isRequired(data.placeName, validation, setValidation, 0, 'place name');
    isRequired(data.description, validation, setValidation, 1, 'description');
    isNumeric(data.latitude, validation, setValidation, 2, 'latitude');
    isNumeric(data.longitude, validation, setValidation, 3, 'longitude');
    isRequired(distance, validation, setValidation, 4, 'distance');
    isRequiredArr(tmethods, validation, setValidation, 5, 'method');
    isRequired(timetoreach, validation, setValidation, 6, 'time to reach');

    if (!validation.validation[0] && !validation.validation[1] && !validation.validation[2] && !validation.validation[3] && !validation.validation[4]
      && !validation.validation[5]) {
      return true;
    }
    return false;
  }
  const handleSubmit = async (event) => {
    event.preventDefault();
    var isValid = validateBeforeSubmit();
    if (isValid) {
      var myHeaders = new Headers();
      myHeaders.append("Content-Type", "application/json");
      var { Authorization, token } = authHeader();
      myHeaders.append(Authorization, token);
      var dataToSend = {
        name: data.placeName,
        description: data.description,
        location: {
          coordinates: [data.longitude, data.latitude]
        },
        distance: distance,
        methods: tmethods,
        timeToReach: timetoreach,
        images: []
      }
      try {
        var raw = JSON.stringify(dataToSend);

        var requestOptions = {
          method: 'POST',
          headers: myHeaders,
          body: raw,
          redirect: 'follow'
        };

        fetch(`${BASE_URL}/places`, requestOptions)
          .then(response => response.json())
          .then(result => showResult(result))
          .catch(error => console.log('error', error));
      } catch (e) {
        console.error(e);
      } finally {
        console.log('We do cleanup here');
      }

    }
  };
  // const handleSubmit = (event) => {
  //     event.preventDefault();
  //     var dataToSend = {
  //         name: data.placeName,
  //         description: data.description,
  //         location: {
  //             coordinates: [data.latitude, data.longitude]
  //         },
  //         distance: distance,
  //         methods: tmethods,
  //         timeToReach: timetoreach,
  //         images: []
  //     }
  //     submitData(dataToSend);
  // };
  const submitData = async (data) => {
    visitingPlaceService.addVisitingPlace(data)
      .then((result) => {
        if (result.success) {
          console.log(result.visitingPlace);
        } else {
          // setErrors(result.status);
        }
      });
  }


  const handleChange = (event) => {
    var value = event.target.value;
    var name = event.target.name;

    if (name === 'placeName') {
      isRequired(value, validation, setValidation, 0, 'place name');
    }
    else if (name === 'description') {
      isRequired(value, validation, setValidation, 1, 'description');
    }
    else if (name === 'latitude') {
      isNumeric(value, validation, setValidation, 2, 'latitude')
    }
    else if (name === 'longitude') {
      isNumeric(value, validation, setValidation, 3, 'longitude')
    }
    else if (name === 'distance') {
      isRequired(value, validation, setValidation, 4, 'distance');
    }
    else if (name === 'timeToReach') {
      isRequired(value, validation, setValidation, 5, 'time to reach');
    }

    setData({ ...data, [event.target.name]: value });
  };

  const handleCheckBoxChange = (value) => {
    if (tmethods.includes(value)) {
      const index = tmethods.indexOf(value);
      if (index > -1) {
        tmethods.splice(index, 1);
      }
      var methods = tmethods;
      setTmethods(methods);
    } else {
      var methods = tmethods.concat(value);
      setTmethods(methods);
    }
  }

  const handleClickonMap = (longitude, latitude) => {
    isNumeric(data.latitude, validation, setValidation, 2, 'latitude');
    isNumeric(data.longitude, validation, setValidation, 3, 'longitude');
    setData({ ...data, latitude: latitude, longitude: longitude });
    console.log(latitude, longitude);
  }
  return (

    <Box my="4" border="1px" borderColor="gray.200" p={['5px', '15px', '25px']}>
      <Heading as="h4" size="md" textAlign='center' pb='20px' pt='5px'>
        Add a Visiting Place
            </Heading>
      {hasError ? <Alert status="error" mb='20px' >
        <AlertIcon />
                Error
            </Alert> : null}
      <form action='submit'>
        <Stack spacing={3}>
          <Text >Name of the place</Text>
          <FormControl isRequired>
            <Input
              data-testid = "placename"
              type='text'
              value={data.placeName}
              name='placeName'
              placeholder='Name of the place'
              aria-label='Name'
              onChange={(event) => handleChange(event)}
              isInvalid={validation.validation[0]}
              errorBorderColor="crimson"
            />
            <FormHelperText color='crimson'>{validation.errorMessage[0]}</FormHelperText>
          </FormControl>
          <Text >Description</Text>
          <FormControl isRequired>
            <Textarea
            data-testid = "description"
              type='text'
              name='description'
              value={data.description}
              placeholder='Description'
              aria-label='Description'
              onChange={(event) => handleChange(event)}
              isInvalid={validation.validation[1]}
              errorBorderColor="crimson"
            />
            <FormHelperText color='crimson'>{validation.errorMessage[1]}</FormHelperText>
          </FormControl>
          <Divider />




          <Text >Location</Text>
          <FormControl isRequired>
            <InputGroup>
              <Input
              data-testid = "latitude"
                type='text'
                name='latitude'
                value={data.latitude}
                placeholder='Latitude'
                aria-label='Latitude'
                onChange={(event) => handleChange(event)}
                isInvalid={validation.validation[2]}
                errorBorderColor="crimson"
              />
              <Input
              data-testid = "longitude"
                type='text'
                name='longitude'
                value={data.longitude}
                placeholder='Longitude'
                aria-label='Longitude'
                onChange={(event) => handleChange(event)}
                isInvalid={validation.validation[3]}
                errorBorderColor="crimson"
              />
            </InputGroup>
            <FormHelperText color='crimson'>{validation.errorMessage[2]} {validation.errorMessage[3]}</FormHelperText>
          </FormControl>
          <Map onClickonMap={handleClickonMap} />



          <Text >Distance (Km)</Text>
          <FormControl isRequired>
            <NumberInput step={0.100} defaultValue={0} min={0.100} max={30} precision={3}
              isInvalid={validation.validation[4]}
              onChange={(valueString) => setDistance(valueString)}
              value={distance}
              errorBorderColor="crimson" >
              <NumberInputField
                placeholder='Distance'
                name='distance'
              />
              <NumberInputStepper>
                <NumberIncrementStepper />
                <NumberDecrementStepper />
              </NumberInputStepper>
            </NumberInput>
            <FormHelperText color='crimson'>{validation.errorMessage[4]}</FormHelperText>
          </FormControl>

          <Text >Available Travelling Methods</Text>
          <FormControl isRequired>
            <Stack direction="row">
              <Checkbox
                onChange={(e) => handleCheckBoxChange('Bus')}
              >
                Bus
                            </Checkbox>
              <Checkbox
                onChange={(e) => handleCheckBoxChange('Train')}
              >
                Train
                                </Checkbox>
              <Checkbox
                onChange={(e) => handleCheckBoxChange('Car')}
              >
                Car
                                </Checkbox>
              <Checkbox
                onChange={(e) => handleCheckBoxChange('Threewheeler')}
              >
                Threewheeler
                                </Checkbox>
              <Checkbox
                onChange={(e) => handleCheckBoxChange('Foot')}
              >
                Foot
                                </Checkbox>

            </Stack>
            <FormHelperText color='crimson'>{validation.errorMessage[5]}</FormHelperText>
          </FormControl>

          <Text >Time to reach (min)</Text>
          <FormControl isRequired>
            <NumberInput step={1} defaultValue={0} min={1} max={180}
              isInvalid={validation.validation[6]}
              errorBorderColor="crimson"
              onChange={(valueString) => setTimetoreach(valueString)}
              value={timetoreach} >
              <NumberInputField
                placeholder='Time To Reach'
                name='timeToReach'
              />
              <NumberInputStepper>
                <NumberIncrementStepper />
                <NumberDecrementStepper />
              </NumberInputStepper>
            </NumberInput>
            <FormHelperText color='crimson'>{validation.errorMessage[6]}</FormHelperText>
          </FormControl>
          <Text >Add Images</Text>

          <Button colorScheme='cyan' type="submit" onClick={(event) => handleSubmit(event)}>Add the place</Button>

        </Stack>
      </form>

      <Modal isOpen={isOpen} onClose={onClose} isCentered>
        <ModalOverlay />
        <ModalContent>
          <ModalHeader>Modal Title</ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <Center>
              {!confirm ?
                <VStack mb='10px'>
                  <CheckCircleIcon w={10} h={10} color="green.500" />
                  <Text>Added Sucessfully</Text>
                  <Button><Link to="/visitingplaces" >Return</Link></Button>
                </VStack>
                :
                <VStack>
                  <QuestionIcon w={10} h={10} color="red.500" />
                  <Text>Unsuccessful</Text>
                </VStack>
              }
            </Center>

          </ModalBody>


        </ModalContent>
      </Modal>
    </Box>

  );
}

export default VisitingPlaceForm;