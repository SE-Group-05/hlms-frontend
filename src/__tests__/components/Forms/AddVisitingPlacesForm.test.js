// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import AddVisitingPlacesForm from "../../../components/Forms/AddVisitingPlacesForm";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <AddVisitingPlacesForm data={{ placename: "Galle", methods: ["bus"] }} />,
    div
  );
});


it("add place correctly", () => {
  const { queryByTestId, getByText } = render(
    <AddVisitingPlacesForm />
  );

 
  
  fireEvent.click(getByText("Bus")) 
  fireEvent.click(getByText("Car")) 
  fireEvent.click(getByText("Foot")) 


  fireEvent.change(queryByTestId("placename"),{ target: { value: 'mirissa' } })
  fireEvent.change(queryByTestId("latitude"),{ target: { value: '23' } })
  fireEvent.change(queryByTestId("longitude"),{ target: { value: '23' } })
  fireEvent.change(queryByTestId("description"),{ target: { value: 'agasghasg hasdh' } })

 
  fireEvent.click(getByText("Add the place")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <AddVisitingPlacesForm />
  );

  expect(tree).toMatchSnapshot();
});
