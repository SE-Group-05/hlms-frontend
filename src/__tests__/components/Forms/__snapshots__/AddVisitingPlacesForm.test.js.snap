// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-19d9m9t"
>
  <h4
    className="chakra-heading css-biiv2x"
  >
    Add a Visiting Place
  </h4>
  <form
    action="submit"
  >
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <p
        className="chakra-text css-0"
      >
        Name of the place
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <input
          aria-describedby="field-98-helptext"
          aria-label="Name"
          aria-required={true}
          className="chakra-input css-0"
          data-testid="placename"
          disabled={false}
          id="field-98"
          name="placeName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Name of the place"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-98-helptext"
        >
          
        </div>
      </div>
      <p
        className="chakra-text css-0"
      >
        Description
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <textarea
          aria-describedby="field-99-helptext"
          aria-label="Description"
          aria-required={true}
          className="chakra-textarea css-0"
          data-testid="description"
          disabled={false}
          id="field-99"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Description"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-99-helptext"
        >
          
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
        role="separator"
      />
      <p
        className="chakra-text css-0"
      >
        Location
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="chakra-input__group css-4302v8"
        >
          <input
            aria-describedby="field-100-helptext"
            aria-label="Latitude"
            aria-required={true}
            className="chakra-input css-0"
            data-testid="latitude"
            disabled={false}
            id="field-100"
            name="latitude"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Latitude"
            readOnly={false}
            required={true}
            type="text"
            value=""
          />
          <input
            aria-describedby="field-100-helptext"
            aria-label="Longitude"
            aria-required={true}
            className="chakra-input css-0"
            data-testid="longitude"
            disabled={false}
            id="field-100"
            name="longitude"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Longitude"
            readOnly={false}
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-100-helptext"
        >
          
           
          
        </div>
      </div>
      <div
        className="css-0"
        data-testid="map"
      >
        <div
          style={
            Object {
              "cursor": "grab",
              "height": 400,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "height": "100%",
                "position": "relative",
                "width": "100%",
              }
            }
          >
            <div
              className=""
              style={
                Object {
                  "height": "100%",
                  "overflow": "hidden",
                  "position": "absolute",
                  "visibility": "inherit",
                  "width": "100%",
                }
              }
            />
          </div>
        </div>
      </div>
      <p
        className="chakra-text css-0"
      >
        Distance (Km)
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="css-e69dqy"
          defaultValue={0}
          value={0}
        >
          <input
            aria-describedby="field-101-helptext"
            aria-disabled={false}
            aria-invalid={true}
            aria-required={true}
            aria-valuemax={30}
            aria-valuemin={0.1}
            aria-valuenow={0}
            aria-valuetext="0"
            autoComplete="off"
            autoCorrect="off"
            className="css-8atqhb"
            disabled={false}
            id="field-101"
            inputMode="decimal"
            name="distance"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            pattern="[0-9]*(.[0-9]+)?"
            placeholder="Distance"
            readOnly={false}
            required={true}
            role="spinbutton"
            type="text"
            value={0}
          />
          <div
            aria-hidden={true}
            className="css-b10tx0"
          >
            <div
              className="css-1ob8896"
              disabled={false}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex={-1}
            >
              <svg
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              className="css-1ob8896"
              disabled={false}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex={-1}
            >
              <svg
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-101-helptext"
        >
          
        </div>
      </div>
      <p
        className="chakra-text css-0"
      >
        Available Travelling Methods
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="chakra-stack css-loe1j3"
        >
          <label
            className="chakra-checkbox css-dfdulk"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Bus
            </span>
          </label>
          <label
            className="chakra-checkbox css-dfdulk"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Train
            </span>
          </label>
          <label
            className="chakra-checkbox css-dfdulk"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Car
            </span>
          </label>
          <label
            className="chakra-checkbox css-dfdulk"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Threewheeler
            </span>
          </label>
          <label
            className="chakra-checkbox css-dfdulk"
          >
            <input
              checked={false}
              className="chakra-checkbox__input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              style={
                Object {
                  "border": "0px",
                  "clip": "rect(0px, 0px, 0px, 0px)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0px",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
              type="checkbox"
            />
            <span
              aria-hidden={true}
              className="chakra-checkbox__control css-dnty2r"
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
            />
            <span
              className="chakra-checkbox__label css-1oeb2oe"
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              Foot
            </span>
          </label>
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-102-helptext"
        >
          
        </div>
      </div>
      <p
        className="chakra-text css-0"
      >
        Time to reach (min)
      </p>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          className="css-e69dqy"
          defaultValue={0}
          value={0}
        >
          <input
            aria-describedby="field-103-helptext"
            aria-disabled={false}
            aria-invalid={true}
            aria-required={true}
            aria-valuemax={180}
            aria-valuemin={1}
            aria-valuenow={0}
            aria-valuetext="0"
            autoComplete="off"
            autoCorrect="off"
            className="css-8atqhb"
            disabled={false}
            id="field-103"
            inputMode="decimal"
            name="timeToReach"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            pattern="[0-9]*(.[0-9]+)?"
            placeholder="Time To Reach"
            readOnly={false}
            required={true}
            role="spinbutton"
            type="text"
            value={0}
          />
          <div
            aria-hidden={true}
            className="css-b10tx0"
          >
            <div
              className="css-1ob8896"
              disabled={false}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex={-1}
            >
              <svg
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              className="css-1ob8896"
              disabled={false}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              role="button"
              tabIndex={-1}
            >
              <svg
                className="chakra-icon css-onkibi"
                focusable={false}
                viewBox="0 0 24 24"
              >
                <path
                  d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-103-helptext"
        >
          
        </div>
      </div>
      <p
        className="chakra-text css-0"
      >
        Add Images
      </p>
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="submit"
      >
        Add the place
      </button>
    </div>
  </form>
</div>
`;
